### ISSUE TITLE
`src` pattern DOES NOT ignore elements

### ISSUE THREAD

**Issue by oncletom** [oncletom/grunt-crx#44]
This is pretty serious because it means it does *ignore* the ignore thing…

```js
grunt.initConfig({
  crx: {
    myHostedPackage: {
      "src": [
        "path/to/**/*",
        "!private-content"
      ],
      // ...
    }
  }
})
```

The file is not going to be ignored and will be included in the crx/zip package.

Two reasons why:

- `grunt-crx` does do the file copy anymore (I think, my memory might be flaky on this point)
- `crx` accepts a folder location as an input to pack/zip so it does blindly trust its content

Two solutions come out of my head:

- creating a temp folder with grunt-crx, copying by respecting `grunt-crx` and then delegate to crx the rest of the packing (might result in a double copy, which is not i/o/hard drive efficient)
- making sure that `crx` uses an array of glob patterns (as grunt does) to move the logic down to it, and to make `grunt-crx` really only an orchestration on top of `crx`.

v0.3.4 has been tagged as the `latest` version to avoid security issues in packaged extensions.

**Comment by Redsandro** [comment]
There are a lot of refs and the refs have a lot of refs. Not easy to grasp for someone passing by. Could you summarize the full problem in your first post?

**Comment by oncletom** [comment]
You are right @Redsandro. I amended the issue description. Let me know if it makes more sense now.

**Comment by Redsandro** [comment]
I get the global problem now. Assuming you mean _`grunt-crx` does_ __not__ _do the copying anymore_, I understand the problem is that [`crx`](https://github.com/oncletom/crx) does simply work on a singe source directory without any extra processing.

I don't mean to ask stupid questions.. but I kinda have to ask.

* Why would `grunt-crx` need to be in a temp folder?
* Why would this mean _"might result in a double copy, which is not i/o/hard drive efficient"_
* Just copying the `src` respecting `excludes` to `tmp` and offering this `tmp` to [`crx`](https://github.com/oncletom/crx) seems to be the shortest bridge. 
  * I think having `1.0` in the `npm` repo is more valuable than avoid having a `tmp` directory (for one because it also saves `.zip` which is necessary for webstore updating anyway).

The [`copy-to`](https://github.com/clavery/grunt-copy-to/blob/12f717869c13f9c02ec3948799155c2eeaf1faae/tasks/copy_to.js) module uses `grunt.file` to do this:

```
this.files.forEach(function(pair) {
	pair.src.forEach(function(src) {
		// (...)
		if (grunt.file.isMatch(options.ignore, src)) {
			grunt.verbose.writeln('Ignored ' + src.red);
			filesIgnored++;
		} else {
			// (...)
			grunt.file.copy(fullpath, dest, copyOptions);
```
